public/blog/2023-11-dask-survival-analysis/2023-11-dask-survival-analysis.md

```diff
@@ -0,0 +1,48 @@
+
+<WindowScreenshot src="gepia-example.png" />
+
+In their very essence, these tools allow you to enter a gene name and output the results of the analysis on pre-curated TCGA datasets. For example, in the image above, I have entered the gene name TP53 and selected the cancer type lower-grade glioma (TCGA-LGG). The output is the Kaplan-Meier plot for the high- and low-risk patient groups. I can also see the p-value of the log-rank test, which tells me how significant the difference in survival between the groups is. For a detailed description of functionalities, I suggest you read this review paper by [Zheng et al.](https://www.frontiersin.org/journals/oncology/articles/10.3389/fonc.2020.00068/full).
+
+
+Perfect, this is just what I need. So where is the problem? Well, as it turns out, you can't just blindly trust the underlying source of survival data of those tools. Curated datasets between tools can vary (more in [this paper](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8726696/) by Idogawa et al.). And this is no small thing. How can you trust the analysis if you can trust the data source? Adding fuel to the fire here, I will let the reader guess how many of those tools are open-sourced.
+
+Ok, don't get me wrong. The utility of such tools is great despite the limitations I mentioned. However, I would like to analyze data from sources I trust. For that, I have prepared [notebooks](https://github.com/jakakokosar/tcga-data) that I can use to get gene expression and corresponding survival data for all cancer types in the TCGA project. I see the irony here. I am not asking you to trust my code. For those who read the code, you verify and replicate it on your own. For others, use the official [GDC](https://portal.gdc.cancer.gov/) data portal to download data.
+
+The combined dataset from TCGA consists of approximately 10K samples and 60K gene expression values. The size of the file on my disk is around 3GB. You might think this is not much, but try loading it in Pandas. But we did not come this far just to use code again. Can we use any existing visual analytics tools?
+
+Kaplan-Meier plotter stands out because it allows me to upload my own data (a rare find among all the other tools). That is just what I was looking for!
+
+<WindowScreenshot src="km-plotter-upload.png" />
+
+The option to upload my own data is cool, but there is a 4MB limitation.
+
+What about Orange? 
+
+<WindowScreenshot src="orange-not-responding.png" />
+
+Well... let's say that things quickly get out of hand. I am going to refer you to [Marko's blog](https://orangedatamining.com/blog/dask-spectroscopy/), where he explains where the problem is. There, Marko also explains our recent efforts to enable [Orange](https://github.com/biolab/orange3/wiki/Orange-with-Dask) for large datasets by integrating Dask.
```

Comment:
" quickly get out of hand": You wrote about 4MB limitation of the other tool, what would be a limit for base Orange. A few 100 MB would probably work? Perhaps try to bit a bit more specific - now this can be read as if Orange can not do anything. :)

Reply:
Of course you can refer to other blog posts, but please quickly summarize it, like 2-3 sentences.